lib/core/scheduler.rb

```diff
@@ -15,26 +15,41 @@
 
 module Core
   class Scheduler
-    def initialize(connector_id, poll_interval)
-      @connector_id = connector_id
+    attr_reader :is_shutting_down
+
+    def initialize(poll_interval)
       @poll_interval = poll_interval
+      @is_shutting_down = is_shutting_down
```

Comment:
From the dispatcher's `shutdown` method. Also, it's used in tests.

Reply:
I don't see `is_shutting_down` is passed in anywhere.

Reply:
`dispatcher.shutdown` calls `scheduler.shutdown`.
The variable is used in the cycle to check if it should loop again.
The variable is also set in tests - `native_scheduler_spec`.
```
    def shutdown
      Utility::Logger.info("Shutting down dispatcher with pool [#{@pool&.class}] and scheduler [#{@scheduler&.class}]...")
      @is_shutting_down = true
      @scheduler&.shutdown
      @pool.shutdown
      @pool.wait_for_termination(TERMINATION_TIMEOUT)
    end
```

Reply:
I think it should be initialized to `false` in method `initialize`:
```
@is_shutting_down = false
```

Otherwise `is_shutting_down` is not defined here?